### PR TITLE
Da encrypt url

### PR DESCRIPTION
**Pull request by danamansana**
This branch adds encrypted urls for Google API. Running locally now should not require getting any secrets, and deploying will not require API_ROOT to be set by hand

### CODE REVIEW

**Comment by ktp242** [comment]
The base URL should be `pr_approve`. `master` will be deploy to production directly.

File: README.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ From the CLI, run `npm run lint` to run the ESLint tool for checking JavaScript
 We use Webpack to fire off a hot-reloading development server. This allows for continuous code changes without the need to refresh your browser.
 
 To run locally, run:
-`API_ROOT='ENTER_URL_WITH_CREDENTIALS_HERE' node index`. You will have to change the string `ENTER_URL_WITH_CREDENTIALS_HERE` in the 'local' script in package.json to the actual API endpoint. Leave the quotation marks. Get this from a coworker.
+`npm run local`
```
Comment (ktp242) [inline README.md]: Can we just assign this script to `npm start`

File: webpack.config.js [hunk 2]
```diff
@@ -14,6 +16,12 @@ const ENV = process.env.NODE_ENV || 'development';
 // Sets appEnv so the the header component will point to the search app on either Dev or Prod
 const appEnv = process.env.APP_ENV ? process.env.APP_ENV : 'production';
 
+// set API_ROOT to the correct encrypted value
+process.env.API_ROOT = appEnv === 'development' || process.env.AWS_PROFILE === 'nypl-sandbox' ? development : production;
```
[the remark this answers is not in the document]
Comment (danamansana) [reply webpack.config.js]: I've set the default local account to nypl-digital-dev, so we can eliminate the || proceess.env.AWS_PROFILE ... ,
That simplifies the logic a bit but I think we need all three variables. AWS_PROFILE is used by aws-sdk to decrypt API_ROOT, API_ROOT is used by google.  APP_ENV is used to determine all the other things that differ between development and production, e.g. the header component

File: README.md (hunk 1 above)
Comment (danamansana) [reply README.md]: I ran into a problem with that earlier because travis runs npm start.

**Comment by danamansana** [comment]
Changed the base and added some comments for kms-helper

File: README.md (hunk 1 above)
Comment (ktp242) [reply README.md]: Maybe I can try investigating this? Where do we tell Travis to run `npm start`? I didn't see it in `.travis.yml`.

Also if we update the way to start the app locally, we might have to update `eb start` as well.

File: webpack.config.js [hunk 3]
```diff
@@ -3,7 +3,9 @@ const webpack = require('webpack');
 const merge = require('webpack-merge');
 const cleanBuild = require('clean-webpack-plugin');
 const ExtractTextPlugin = require('extract-text-webpack-plugin');
+const aws = require('./lib/kms-helper.js');
 const pkg = require('./package.json');
+const { api: { development, production } } = require('./urlConfig.js');
```
Comment (ktp242) [inline webpack.config.js]: I feel it could have better naming for `urlConfig`

Also, now the variables `development` and `production` stands for two encrypted urls. I don't think their name reflect the fact clear enough. `development` and `production` sound to stand for environments (and they should be).

File: appConfig.js [hunk 4]
```diff
@@ -4,21 +4,4 @@ export default {
   favIconPath: '//d2znry4lg8s0tq.cloudfront.net/images/favicon.ico',
   port: 3001,
   webpackDevServerPort: 3000,
-  api: {
-    root: {
```
Comment (ktp242) [inline appConfig.js]: Why don't we just put the codes in `urlConfig` in `appConfig.js`.

File: webpack.config.js [hunk 5]
```diff
@@ -14,6 +16,12 @@ const ENV = process.env.NODE_ENV || 'development';
 // Sets appEnv so the the header component will point to the search app on either Dev or Prod
 const appEnv = process.env.APP_ENV ? process.env.APP_ENV : 'production';
 
+// set API_ROOT to the correct encrypted value
+process.env.API_ROOT = appEnv === 'development' ? development : production;
```
Comment (ktp242) [inline webpack.config.js]: I don't think we should mutate `process.env.API_ROOT` in our code. The idea of `process.env` if I understand it correctly is to serve as the fundamental state to present how an app should config in a certain context(environment). 

The fact we change `process.env.API_ROOT` indicate that it is based on the other config, thus it doesn't need to be an env var.

File: webpack.config.js (hunk 3 above)
Comment (danamansana) [reply webpack.config.js]: OK I can change the name-- perhaps to developmentUrl and productionUrl?

File: appConfig.js (hunk 4 above)
Comment (danamansana) [reply appConfig.js]: I ran into an issue with exporting: webpack makes import/export available in other files but not in the webpack config itself, where you have to use javascript's module.exports/require pattern. Unfortunately these require slightly different structures for the file.

File: webpack.config.js (hunk 5 above)
Comment (danamansana) [reply webpack.config.js]: OK, what do you think is the best way to store this value, then?

File: README.md (hunk 1 above)
Comment (danamansana) [reply README.md]: Hmmm good point not sure I understand why I was getting a problem with travis. As for eb start I think it's fine -- nothing on AWS should require AWS_PROFILe to be set, which the only difference between the local and start commands

**Comment by ktp242** [comment]
I ran into this error when I tried to run this branch locally

```
UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): AccessDeniedException: The ciphertext refers to a customer master key that does not exist, does not exist in this region, or you are not allowed to access.
error calling API : Error: socket hang up
```

**Comment by danamansana** [comment]
Weird. I only see change to package-lock and .gitignore. I've pushed them up anyway though, and it's working for me locally. Let me know if you are still seeing this problem

File: src/server/ApiRoutes/cacheUtil.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import redis from 'redis';
```
Comment (ktp242) [inline src/server/ApiRoutes/cacheUtil.js]: Shouldn't this update be another PR?

Comment (danamansana) [reply src/server/ApiRoutes/cacheUtil.js]: Good catch, not sure how that got in there
